processing/src/main/java/org/apache/druid/query/QueryDataSource.java

```diff
@@ -97,9 +98,6 @@ public boolean isCacheable(boolean isBroker)
   @Override
   public boolean isGlobal()
   {
-    // FIXME: this was query.getDataSource().isGlobal()
-    // but it feeled dodgy to me
-    // in general: I think isProcessable should be a precondition to isGblobal
     return query.getDataSource().isGlobal();
```

Comment:
This (and the `isProcessable()` impl as well) does still feel sketchy, but we can leave it as-is for now. Could you please include a comment about that, though? Maybe something like:

```
// It would be more accurate if this method returned false, but returning false
// currently causes XYZ test case to fail.
```

(& mention the test case)

Reply:
actually; with the most recent changes - I think this would not cause any difference anymore ; updated to return `false`

the tests which needed modifications are in [this commit](https://github.com/apache/druid/pull/17763/commits/f28a5aa28b46ea7ca69784d13fe78c609263e7c6)